var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -105,7 +105,7 @@ class Opencv(CMakePackage, CudaPackage):
     # the current development branch of OpenCV. See #8461 for more information.
     patch('dnn_cuda.patch', when='@3.3.0:3.4.1+cuda+dnn')
 
-    depends_on('eigen~mpfr', when='+eigen', type='build')
+    depends_on('eigen', when='+eigen', type='build')
```

Comment:
Can you remove the `type=build` as well? As weird as it sounds in our current dependency model header only libraries are `build,link` which is the default.

Reply:
:eyes: Good eye, removed!